komodo/build.py

```diff
@@ -116,6 +120,50 @@ def rsync(pkg, ver, pkgpath, data, prefix, *args, **kwargs):
     )
 
 
+def download(pkg, ver, pkgpath, data, prefix, *args, **kwargs):
+    print(f"Installing {pkg} ({ver}) with download")
+
+    url = kwargs["url"]
+    if not url.startswith("https"):
+        raise ValueError(f"{url} does not use https:// protocol")
+
```

Comment:
If `hash` is required then you probably need to 
```suggestion
if not "hash" in kwargs:
    raise ValueError("no hash provided")
```

Reply:
This is almost exactly equal to the KeyError, so this is up to you.